### DISCUSSION TITLE
Kudos to PyMuPDF!

### DISCUSSION THREAD

**Discussion by gamesbook** [pymupdf/PyMuPDF#3794] [Show and tell]
As always happens, I am again blown away by the awesome libraries provided by the Python community!  

I am working on an app to create various types of graphics, based on ReportLab (an excellent tool!) and needed a way to also get a image output.  Within less than an hour after coming across a link to PyMUPDF, I was able to add this function to my app and *boom* instant PNG files.
```python
def pdf_to_png(filename: str, dpi: int = 300):
    """Extract pages from PDF as PNG image(s)."""
    feedback(f'Saving content of {filename} as .png image(s)...')
    basename = os.path.splitext(filename)[0]
    try:
        doc = pymupdf.open(filename)
        pages = doc.page_count
        for page in doc:
            pix = page.get_pixmap(dpi=dpi)
            if pages > 1:
                pix.save(f"{basename}_{page.number + 1}.png")
            else:
                pix.save(f"{basename}.png")
    except Exception as err:
        feedback(f'Unable to extract images for {filename} - {err}!')

```
PyMuPDF is **really** easy to use and "just works" - brilliant!

**Comment by JorjMcKie** [comment]
Really nice feedback - thank you!

You won't be surprised to read that your code just scratched the surface, will you 😉?
Maybe the right point in time to interest you in our own specialized AI, called "Dr. PDF". Access it via https://pymupdf.io/.
On the left hand side you find interactive terminal windows, the right hand side is reserved for that chatbot. The PyMuPDF maintainers have spent significant effort to make it precise and exhaustive in terms of covering features relevant for any newcomer ...

**Reply by gamesbook** [reply]
Even more awesome - will take a look, thanks!

(**PS** - brilliant user name - from some of his really under-appreciated novels)
